src/app/shared/case-editor/case-edit-page.component.ts

```diff
@@ -72,7 +73,8 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
       .filter(caseField => !this.caseFieldService.isReadOnly(caseField))
       .every(caseField => {
         let theControl = this.editForm.controls['data'].get(caseField.id);
-        return this.checkDocumentField(caseField, theControl) && this.checkOptionalField(caseField, theControl);
+        return this.isHidden(caseField, this.editForm.getRawValue())
```

Comment:
there's a lot of complexity growing here. If you have time @tzarsmango, it would be good to move all the logic of the method currentPageIsNotValid into a separate service PageValidationService

Reply:
👍